src/js/utils/styles.js

```diff
@@ -208,18 +234,56 @@ export const focusStyle = css`
   }
 `;
 
+export const getInputPadBySide = (props, side) => {
+  if (typeof props.theme.global.input.padding !== 'object') {
+    return props.theme.global.input.padding;
+  }
+
+  let orientation;
+  if (side === 'left' || side === 'right') orientation = 'horizontal';
+  else if (side === 'top' || side === 'bottom') orientation = 'vertical';
+  else orientation = undefined;
+
+  // if individual side isn't available, fallback to the
+  // orientation if possible
+  const pad =
+    props.theme.global.input.padding[side] ||
+    props.theme.global.input.padding[orientation];
+
+  // for backwards compatibility we need to add back the control border width
+  // in each case, check first if the value is an edgeSize,
+  // then if it's a string value.
+  const adjustedPad =
+    parseMetricToNum(`${props.theme.global.edgeSize[pad] || pad}px`) +
+    parseMetricToNum(`${props.theme.global.control.border.width}px`);
+
+  // if still undefined, there should be no pad.
+  return `${adjustedPad}px` || '0px';
```

Comment:
This will never be `0px` as the first string will always be `true`.

Reply:
Gotcha, let me remove the check then.

Reply:
still removing?